pkg/autodetect/main_test.go

```diff
@@ -397,102 +397,124 @@ func TestAuthDelegatorBecomesUnavailable(t *testing.T) {
 	b := WithClients(cl, dcl, cl)
 
 	// test
-	b.detectClusterRoles()
+	b.detectClusterRoles(context.Background())
 	assert.True(t, viper.GetBool("auth-delegator-available"))
 
 	cl.CreateFunc = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
 		return fmt.Errorf("faked error")
 	}
-	b.detectClusterRoles()
+	b.detectClusterRoles(context.Background())
 	assert.False(t, viper.GetBool("auth-delegator-available"))
 }
 
 func TestCleanDeployments(t *testing.T) {
-	cl := customFakeClient()
-	cl.CreateFunc = cl.Client.Create
-	dcl := &fakeDiscoveryClient{}
-
-	jaeger1 := v1.NewJaeger(types.NamespacedName{
-		Name:      "TestDeletedInstance",
-		Namespace: "TestNS",
-	})
-
-	jaeger2 := v1.NewJaeger(types.NamespacedName{
-		Name:      "TestDeletedInstance2",
-		Namespace: "TestNS",
-	})
-
-	dep1 := &appsv1.Deployment{
-		Spec: appsv1.DeploymentSpec{
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{},
-				Spec: corev1.PodSpec{
-					Containers: []corev1.Container{
-						corev1.Container{
-							Name:  "C1",
-							Image: "image1",
-						},
-					},
-				},
-			},
+	for _, tt := range []struct {
+		cap             string // caption for the test
+		watchNamespace  string // the value for WATCH_NAMESPACE
+		jaegerNamespace string // in which namespace the jaeger exists, empty for non existing
+		deleted         bool   // whether the sidecar should have been deleted
+	}{
+		{
+			cap:             "existing-same-namespace",
+			watchNamespace:  "observability",
+			jaegerNamespace: "observability",
+			deleted:         false,
 		},
-	}
-	dep2 := &appsv1.Deployment{
-		Spec: appsv1.DeploymentSpec{
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{},
-				Spec: corev1.PodSpec{
-					Containers: []corev1.Container{
-						corev1.Container{
-							Name:  "C1",
-							Image: "image1",
+		{
+			cap:             "not-existing-same-namespace",
+			watchNamespace:  "observability",
+			jaegerNamespace: "",
+			deleted:         true,
+		},
+		{
+			cap:             "existing-watched-namespace",
+			watchNamespace:  "observability,other-observability",
+			jaegerNamespace: "other-observability",
+			deleted:         false,
+		},
+		{
+			cap:             "existing-non-watched-namespace",
```

Comment:
The caption is not unique. The same caption is used in the following test

Reply:
Done